httpbp/middlewares.go

```diff
@@ -246,3 +247,35 @@ func SupportedMethods(method string, additional ...string) Middleware {
 		}
 	}
 }
+
+// recoverPanic recovers from any panics, logs them, and sets the returned error
+// to a generic 500 error. recoverPanic is always the last middleware in the
+// middleware chain, so it is the first one when returning which lets the error
+// bubble up into other middlewares. Since it is always added to the middleware
+// chain is a specific position, it is not exported.
+func recoverPanic(name string, next HandlerFunc) HandlerFunc {
+	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) (err error) {
+		defer func() {
+			if r := recover(); r != nil {
+				var rErr error
```

Comment:
Actually, why do we need to convert this to an error?  "endpoint" is logged as a keyval, so adding the "panic in %q" is redundant.  Can we not just do

```
log.C(ctx).Errorw("recovered from panic",
  "panic", r,
  "endpoint", name,
)
```

Presumably the logger will stringify or jsonify properly?

Reply:
it will try to auto jsonify it for `interface{}`, which might not be the desired outcome if it's an error (which would be the majority of the errors), so I think trying to convert it into an error can still be useful. we could use `r` directly if it's not an error, instead of trying to use `fmt.Errorf`, though.

Reply:
We still want it to be an error for when we update the returned `err` value.